File: identhub/src/main/java/de/solarisbank/identhub/BaseFragment.java
package de.solarisbank.identhub;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import de.solarisbank.identhub.di.LibraryComponent;

public abstract class BaseFragment extends Fragment {

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        inject(LibraryComponent.getInstance());
        super.onCreate(savedInstanceState);
    }

    protected abstract void inject(LibraryComponent component);
}

File: identhub/src/main/java/de/solarisbank/identhub/base/BaseFragment.kt
package de.solarisbank.identhub.base

import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import de.solarisbank.identhub.AssistedViewModelFactory
import de.solarisbank.identhub.di.FragmentComponent

abstract class BaseFragment : Fragment() {
    lateinit var assistedViewModelFactory: AssistedViewModelFactory

    lateinit var viewModelFactory: ViewModelProvider.Factory

    override fun onCreate(savedInstanceState: Bundle?) {
        val activityComponent = (requireActivity() as BaseActivity).activityComponent
        inject(FragmentComponent.Initializer.init(activityComponent))
        super.onCreate(savedInstanceState)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initViewModel()
    }

    protected open fun initViewModel() {
        viewModelFactory = assistedViewModelFactory.create(this, arguments)
    }

    protected abstract fun inject(component: FragmentComponent)
}